### PR TITLE
SecurityAgent: syncup from rdkservices:

### PR DESCRIPTION
**Pull request by HaseenaSainul**
RDKDEV-390: SecurityAgent: Secure REST API methods in a granular way
 RDKCOM-3277 RDKDEV-343 - SecurityAgent: token generated is always the same
 DELIA-59688: Upstream security agent patch 231

### CODE REVIEW

File: SecurityAgent/SecurityAgent.h [hunk 1]
```diff
@@ -168,8 +168,9 @@ namespace Plugin {
         std::unique_ptr<TokenDispatcher> _dispatcher; 
         Core::ProxyType<RPC::InvokeServer> _engine;
         string _testtoken;
+        string _servicePrefix;
 
-        static constexpr TCHAR TestTokenContent[] = _T(R"--({ "url": "https://test.url.com", "user":"Test" })--");
```
Comment (pwielders) [inline SecurityAgent/SecurityAgent.h:172]: Should this not alos be enclosed in the SECURITY_TESTING_MODE preprocessor statements ?

**Review by pwielders — CHANGES_REQUESTED** [review]
Message s that indicate expected behaviour, should by definition be TRACES and not LOG lines as LOG lines are messages operators need to act upon. Succesfull things you do not need to at upon.

File: SecurityAgent/SecurityContext.cpp [hunk 2]
```diff
@@ -44,20 +47,35 @@ namespace Plugin {
...
+        return ((_accessControlList != nullptr) && (_accessControlList->Allowed(callsign, method)));
     }
 
     //! Allow a JSONRPC message to be checked before it is offered for processing.
     bool SecurityContext::Allowed(const Core::JSONRPC::Message& message) const /* override */ 
     {
-        return ((_accessControlList != nullptr) && (_accessControlList->Allowed(message.Callsign(), message.Method())));
+        bool bAllowed = ((_accessControlList != nullptr) && (_accessControlList->Allowed(message.Callsign(), message.Method())));
+        if(!bAllowed)
+            SYSLOG(Logging::Notification, ("Access for token url [%s], plugin [%s], method [%s] not allowed", _context.URL.Value().c_str(),message.Callsign().c_str(),message.Method().c_str()));
```
[the remark this answers is not in the document]
Comment (HaseenaSainul) [reply SecurityAgent/SecurityContext.cpp]: done

File: SecurityAgent/SecurityAgent.cpp [hunk 3]
```diff
@@ -119,9 +124,13 @@ namespace Plugin {
         ASSERT(subSystem != nullptr);
 
         string connector = config.Connector.Value();
+
         if (connector.empty() == true) {
             connector = service->VolatilePath() + _T("token");
         }
+
+        SYSLOG(Logging::Notification,(_T("SecurityAgent TokenDispatcher connector path %s"),connector.c_str()));
```
[the remark this answers is not in the document]
Comment (HaseenaSainul) [reply SecurityAgent/SecurityAgent.cpp]: done

File: SecurityAgent/SecurityAgent.h (hunk 1 above)
Comment (HaseenaSainul) [reply SecurityAgent/SecurityAgent.h:172]: done

**Review by pwielders — APPROVED** [review]

**Pull request merged**